src/Common/src/Interop/Windows/mincore/Interop.GetFullPathNameW.cs

```diff
@@ -22,19 +22,25 @@ partial class mincore
         [DllImport(Libraries.CoreFile_L1, EntryPoint = "GetFullPathNameW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
         private static extern int GetFullPathNamePrivate(string path, int numBufferChars, [Out]StringBuilder buffer, IntPtr mustBeZero);
 
-        internal static int GetFullPathName(string path, int numBufferChars, [Out]StringBuilder buffer, IntPtr mustBeZero)
+        internal static int GetFullPathName(string path, int numBufferChars, [Out]StringBuilder buffer)
```

Comment:
> My biggest concern is exposing direct access to the P/Invoke and unwrapped usages slipping through.

Aren't there other cases where some of the raw P/Invokes have been left as internal rather than private?

> Isn't normalizing access to the APIs legitimate?

To a limited degree, sure.  But we're trying to keep these interop files as close to the original as possible, with the few exceptions being places where we want to make small tweaks, such as hiding a particular argument and always filling in the same default value.

> I would think that we would be trying to enforce proper behavior at this level.

At this level, the proper behavior is what the underlying OS provides.  Imagine we had the ability to #include a .h file directly into a C# assembly... at that point the need for these Interop.*.cs files would go away and we'd be left with the PathInternal files that would include this behavior.  We obviously can't just #include them, though, which is why the Interop.*.cs files exist, to provide that level of functionality.

> Why do we allow you to specify a StringBuffer and a size? It would prevent a lot of issues to only expose StringBuffer and get the Capacity in the wrappers.

I don't understand the comparison.  DllImports are able to marshal both strings and StringBuilders as char*, and the runtime can't magically assume that a char* and an int next to each other in a DllImport are related, so the DllImport needs to be written to take both the thing to be marshaled as a char* and the thing to be marshaled as the length.  That said, I'd actually be ok with us making such a DllImport private and exposing as internal a wrapper that did the appropriate delegation to the underlying P/Invoke... that's an example of minimal smoothing over to hide an argument, as I noted above.

> There is also a concern about discovery.

I'm fine if there's a comment in the P/Invoke files stating that the PathInternal functions should be used instead.  Beyond that, I'm also not particularly concerned about discovery as these functions are meant to be used by only a small handful of lower-level assemblies in corefx... anything above those assemblies should be using those assemblies, not the P/Invokes.

> I think of PathInternal as "interop" as well.

In which case, why not have a clean separation between the files that define the P/Invokes to the corresponding native functions and the files that define the wrappers that do the propery path normalization?